### ISSUE TITLE
ChocolatesByNumbers cannot calculate for Bigger N, i.e 1 million 

### ISSUE THREAD

**Issue by mjwahyudi** [ZidaneFirza/Codility#2]
Sesuai dengan soal, range N bisa sangat besar, sampai 1 milyar. Dengan performance diharapkan perhitungan dapat selesai kurang dari 5 detik. Menggunakan test angka besar, contoh N = 1.000.000, test gagal dilakukan.

So far, algoritma yang dibangun sudah cukup baik dan mengcover hampir perhitungan sederhana. Kunci penyelesaian masalah performance untuk perhitungan besar perlu rumus matematika yang mampu menghitung pengulangan yang lebih pendek.

**Comment by mjwahyudi** [comment]
Klo mau tahu rumus matematikanya? hehehehe cari arti kata greatest common divisor dalam bahasa Indonesia. Ini SMP harusnya udah pernah belajar.

**Comment by ZidaneFirza** [comment]
sudah direvisi om, terima kasih

**Comment by mjwahyudi** [comment]
Hahahaha, dapat solusi dari mana? klo bisa nunjukin sumber kenapa n / gcd atau jelasin sendiri, it would be awesom. Either smart being use the google or maybe mathematician genius :p. Good work anyway. Itu bisa lebih rapi yang fungsi while dijadikan recursive.

**Comment by ZidaneFirza** [comment]
Kemarin pas olimpiade sains belajar pake algortima euclid om buat cari gcd

**Comment by mjwahyudi** [comment]
ok sip..

**Comment by ZidaneFirza** [comment]
n / gcd nya maksudnya yg di return sama fungsi gcd om?

**Comment by ZidaneFirza** [comment]
fungsi solusi maksudnya*

**Comment by mjwahyudi** [comment]
Iya, maksudnya dari mana persamaan bisa dapat return N/GCD(M,N). Tapi gpp, whatever it is, its works. Sambil belajar juga unit test ya.

**Comment by ZidaneFirza** [comment]
itu saya ngitung pake coret coretan om nyari gcd nya abis itu cari hubungan
antara hasil, gcd, dan nilai n

**Comment by ZidaneFirza** [comment]
unit test itu yang kayak junit itu ya om?

**Comment by mjwahyudi** [comment]
> unit test itu yang kayak junit itu ya om?
Iyak, JUnit.
